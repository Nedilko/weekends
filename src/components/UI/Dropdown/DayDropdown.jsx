import PropTypes from 'prop-types'
import { getDay, DAYS } from '@utils/convertTime'
import Dropdown from '../Dropdown/Dropdown'

DayDropdown.propTypes = {
  selectedValue: PropTypes.number.isRequired,
  onChange: PropTypes.func.isRequired,
}

function DayDropdown({ selectedValue, onChange }) {
  return (
    <Dropdown
      value={DAYS[selectedValue - 1]}
      items={DAYS}
      label="Day"
      onChange={(value) => onChange(getDay(value))}
    />
  )
}
export default DayDropdown
