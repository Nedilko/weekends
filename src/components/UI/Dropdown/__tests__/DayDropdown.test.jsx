import DayDropdown from '@UI/Dropdown/DayDropdown'
import { render, vi, userEvent } from '@utils/test-utils'
import { screen } from '@testing-library/react'

describe('button', () => {
  beforeAll(() => {
    window.HTMLElement.prototype.scrollIntoView = vi.fn()
  })

  afterEach(() => {
    window.HTMLElement.prototype.scrollIntoView.mockClear()
  })

  it('should render closed', () => {
    const handleChage = vi.fn()
    const selectedValue = 3
    const { container } = render(
      <DayDropdown onChange={handleChage} selectedValue={selectedValue} />
    )
    expect(container).toMatchSnapshot()
  })

  it('should render with label Day', () => {
    const handleChage = vi.fn()
    const selectedValue = 3
    render(<DayDropdown onChange={handleChage} selectedValue={selectedValue} />)
    const label = screen.getByText('Day')
    expect(label).toBeInTheDocument()
  })

  it('should trigger on change handler', async () => {
    const handleChage = vi.fn()
    const selectedValue = 3
    const user = userEvent.setup()
    render(<DayDropdown onChange={handleChage} selectedValue={selectedValue} />)
    const dropdownElement = screen.getByRole('button')
    await user.click(dropdownElement)
    expect(handleChage).toBeCalledTimes(1)
    expect(handleChage).toBeCalledWith(3)
  })
})
