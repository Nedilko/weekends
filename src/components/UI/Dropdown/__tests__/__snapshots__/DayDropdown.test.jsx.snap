// Vitest Snapshot v1

exports[`DayDropdown > should render closed 1`] = `
<div>
  <div
    class="flex flex-row items-center"
  >
    <div
      class="mr-2 flex items-center justify-center"
    >
      Day
    </div>
    <div
      class="z-10 block"
    >
      <button
        class="flex h-6 items-center justify-between rounded-md border border-slate-100 pl-2 text-base font-thin focus:border-slate-300 focus:outline-none dark:border-gray-700 dark:focus:border-gray-500"
        type="button"
      >
        Wednesday
        <svg
          class="mr-1 h-4 w-4 text-gray-500 "
          fill="none"
          role="img"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 9l-7 7-7-7"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`button > should render closed 1`] = `
<div>
  <div
    class="flex flex-row items-center"
  >
    <div
      class="mr-2 flex items-center justify-center"
    >
      Day
    </div>
    <div
      class="z-10 block"
    >
      <button
        class="flex h-6 items-center justify-between rounded-md border border-slate-100 pl-2 text-base font-thin focus:border-slate-300 focus:outline-none dark:border-gray-700 dark:focus:border-gray-500"
        type="button"
      >
        Wednesday
        <svg
          class="mr-1 h-4 w-4 text-gray-500 "
          fill="none"
          role="img"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 9l-7 7-7-7"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
